test/libsolidity/syntaxTests/conversion/function_type_array_to_memory.sol

```diff
@@ -0,0 +1,49 @@
+contract C {
+    function externalDefault() external returns(uint) { return 11; }
+    function externalView() external view returns(uint) { return 12; }
+    function externalPure() external pure returns(uint) { return 13; }
+
+    function internalDefault() internal returns(uint) { return 21; }
+    function internalView() internal view returns(uint) { return 22; }
+    function internalPure() internal pure returns(uint) { return 23; }
+
+    function testViewToDefault() public returns (uint, uint) {
+        function () external returns(uint)[1] memory externalDefaultArray;
+        function () internal returns(uint)[1] memory internalDefaultArray;
+
+        // This would work if we were assigning to storage rather than memory
+        externalDefaultArray = [this.externalView];
+        internalDefaultArray = [internalView];
```

Comment:
Does this work for storage because there elements are copied and converted one by one and the conversion of the whole array is not involved?

Reply:
yes